tempesta_fw/t/unit/test_http_parser.c

```diff
@@ -131,39 +230,58 @@ do_split_and_parse(unsigned char *req_str)
 		 tail_len, (int)tail_len, tail_buf);
 
 	/* We expect that the parser requests more data. */
-	r = tfw_http_parse_req(req, head_buf, head_len);
+	if (type == FUZZ_REQ)
+		r = tfw_http_parse_req(req, head_buf, head_len);
+	else
+		r = tfw_http_parse_resp(resp, head_buf, head_len);
 	if (r != TFW_POSTPONE)
 		return r;
 
 	/* Parse the tail. */
-	return tfw_http_parse_req(req, tail_buf, tail_len);
+	if (type == FUZZ_REQ)
+		return tfw_http_parse_req(req, tail_buf, tail_len);
+	else
+		return tfw_http_parse_resp(resp, tail_buf, tail_len);
```

Comment:
Whole the function and split_and_parse_n() are messy and must be reworked. Is there a real reason why to copy the piece of code

                if (type == FUZZ_REQ)                                          
                        r = tfw_http_parse_req(req, buf, size);          
                else                                                           
                        r = tfw_http_parse_resp(resp, buf, size); 

4 times? While 2 chunks and more are handled differently? Why at the end of do_split_and_parse() we call the parser exactly twice? All the thing make a code reader wondering and must be reworked.